src/domain/enrolment/CreateEnrolmentPage.tsx

```diff
@@ -28,6 +32,21 @@ const CreateEnrolmentPage: React.FC = () => {
   };
 
   const event = eventData?.event;
+  const eventLocationId = event?.location?.id || '';
+  const selectedOccurrrences = Array.isArray(occurrences)
+    ? occurrences
+    : [occurrences];
+
+  const allOccurrences: OccurrenceFieldsFragment[] =
+    event?.pEvent?.occurrences.edges.map(
+      (edge) => edge?.node as OccurrenceFieldsFragment
+    ) || [];
+  const filteredOccurrences: OccurrenceFieldsFragment[] = allOccurrences.filter(
+    (o) => selectedOccurrrences.includes(o.id)
+  );
+
+  const areSelectedOccurrencesValid =
+    event?.pEvent?.neededOccurrences === filteredOccurrences.length;
```

Comment:
neededOccurrences could also come from getEventFields function if added to this component